pages/map.py

```diff
@@ -57,45 +57,78 @@ def create_lines_map(trips_group_by_user_id, user_id_list):
     return fig
 
 
-def create_heatmap_fig(data):
+def get_map_coordinates(trips_group_by_user_mode, user_mode_list):
+    coordinates = {'lat': [], 'lon': [], 'color':[]}
+    for user_mode in user_mode_list:
+        color = trips_group_by_user_mode[user_mode]['color']
+        trips = trips_group_by_user_mode[user_mode]['trips']
+
+        for trip in trips:
+            coordinates['lon'].append(trip['start_coordinates'][0])
+            coordinates['lon'].append(trip['end_coordinates'][0])
+            coordinates['lat'].append(trip['start_coordinates'][1])
+            coordinates['lat'].append(trip['end_coordinates'][1])
+            coordinates['color'].extend([color,color])
+    return coordinates
+
+
+def create_heatmap_fig(trips_group_by_user_mode, user_mode_list):
+    coordinates = get_map_coordinates(trips_group_by_user_mode, user_mode_list)
     fig = go.Figure()
-    if len(data.get('lat', [])) > 0:
+    if len(coordinates.get('lat', [])) > 0:
         fig.add_trace(
             go.Densitymapbox(
-                lon=data['lon'],
-                lat=data['lat'],
+                lon=coordinates['lon'],
+                lat=coordinates['lat'],
+                name = '',
+                
             )
         )
+
+        fig.add_trace(
+            go.Scattermapbox(
```

Comment:
why are we displaying a `Scattermapbox` in `create_heatmap_fig`?
We already have one in `create_bubble_fig` below.

Reply:
And why is this not showing up in the "Testing done"? I checked the heatmaps in the most recent set of tests, and I don't see any bubbles in there. This makes me skeptical of the testing done.

Reply:
I added  it just to differentiate when different modes are selected. I have attached a screenshot of heatmap with bubble above.

Reply:
I am not sure this is useful, and I am not even sure that users will notice the different colored bubbles in the heatmap. I think that just displaying the heatmap that includes all the trips selected is sufficient. @JGreenlee thoughts?

Reply:
Since the heatmap already uses color as an attribute for intensity, we cannot easily convey any other information using color.
I would suggest removing the bubbles from the heatmap.

> I think that just displaying the heatmap that includes all the trips selected is sufficient.

I agree. This is sufficient as long as modes can still be added/removed and the heatmap reflects those changes.

Reply:
@achasmita